test/integration/adoption_application_reviews_test.rb

```diff
@@ -43,16 +43,14 @@ class AdoptionApplicationReviewsTest < ActionDispatch::IntegrationTest
   test "verified staff can edit an adoption application status" do
     sign_in users(:user_two)
 
-    put "/adopter_applications/#{@adopter_application.id}",
-      params: { adopter_application:
-        {
-          status: 'under_review', notes: ''
-        }, commit: 'Save', id: @adopter_application.id
-      }
-
-    assert_response :redirect
-    follow_redirect!
-    assert_select 'div', 'Status: Under Review'
+    assert_changes 'AdopterApplication.find(@adopter_application.id).status' do
+      put "/adopter_applications/#{@adopter_application.id}",
+        params: { adopter_application:
+          {
+            status: 'under_review', notes: ''
+          }, commit: 'Save', id: @adopter_application.id
+        }
```

Comment:
Nice! I have not used assert_changes before. It's a good way to loosely determine a change. Though it does not specifically test the AdopterApplication.status attribute. Is there a way we can assert that? I note there is a from: and to: argument that you can use. Maybe this can be used to assert a change on that attribute?

[assert_changes](https://apidock.com/rails/ActiveSupport/Testing/Assertions/assert_changes) :@object, from: nil, to: :foo do
  @object = :foo
end

If not, I liked your old test that asserted the change, and you could just add a check on the db record by making a query for that record and asserting the status is x?

Reply:
Done! Made it work with `from: , to:` arguments. Glad you noticed that `assert_changes` accepts those arguments! This is definitely the ideal way to write this test

Reply:
Nice that's more specific now! I'll merge.